### ISSUE TITLE
GPU Usage YOLO3 only at about 50 % 

### ISSUE THREAD

**Issue by alexanderfrey** [AlexeyAB/darknet#661]
Hi,

I'm currently trying to increase detection speed on Geforce 1080TI with 11GB. Detection is running on a 4k video stream (3800x800) and the YOLOv3 input layer is width 896 and height 896 to increase detection of smaller objects. The odd thing is that the GPU delivers approx. 9 FPS but has only 50% work load. 

Why is the GPU not running with 100% work load ? Can I increase performance somehow with a second GPU ?

Thanks for any ideas.

Best,
Alexander

**Comment by AlexeyAB** [comment]
Hi,

* Do you use this repository or original?
* What command do you use for detection?
* Do you use video file or network video stream with (3800x800) resolution? If network video-stream, then what FPS and GPU usage cand you get using videofile?

**Comment by alexanderfrey** [comment]
Hi,

I’m using this repo and I use the python bindings (darknet.py) for detection from the darknet repo. I read the frames from a local mp4 file in a while loop and execute detection with 

r = detect(net, meta, "data/dog.jpg")

Now that you asked I wonder if I introduced the problem myself by using those python bindings. I don’t know about yolo2 but I can check once I’m back at my computer.

Thanks for suggestions. 

Best
Alexander

**Comment by alexanderfrey** [comment]
Small remark: Instead of the static image dog.jpg I use the frame extracted from the video file.

**Comment by AlexeyAB** [comment]
Just try to run:

* `darknet.exe detector demo data/coco.data cfg/yolov3.cfg yolov3.weights test.mp4`

* `darknet.exe detector demo data/coco.data cfg/yolov2.cfg yolov2.weights test.mp4`

How many FPS can you get, and how many % of the GPU usage?

**Comment by alexanderfrey** [comment]
Hi,

thanks for your help. Very much appreciated !

I tried out with the video_yolo3.sh script and it indeed runs much faster with approx. 20fps and GPU usage is also around 80%. But detection is much worse.

From what I see I can tell that running with the script the video seems to be downscaled now and the GPU memory consumption is only at about 5GB compared to the python binding variant which is at 10 GB. I guess this might be the large images loaded into the gpu memory ? But why does it not fully load the images into the gpu now ?

I tested with the same video file of course and with the same yolov3.cfg.

Thank you again,
Alexander

**Comment by alexanderfrey** [comment]
Small remark: The detection with the script gets only worse when tinkering the -thresh parameter. (Default seems to be 0.25)

**Comment by AlexeyAB** [comment]
> I also noticed that darknet.py references lib = CDLL("~/darknet/libdarknet.so", RTLD_GLOBAL)
> while in your code you reference yolo_cpp_dll.dll
> I'm on Linux here, so can I still compile and use the yolo_cpp_dll.dll ?

Yes, you can, set in the Makefile `GPU=1 CUDNN=1 LIBSO=1`, do `make -j8`
then run `darknet.py` using Python 2.x - it will use `darknet.so` on Linux (that is the same as `yolo_cpp_dll.dll` on Windows)
